### PR TITLE
[AT-1456] upgrade flink/bom dependency to use latest versions

### PR DESCRIPTION
**Pull request by dkovalenkoI**
## Summary
Flink application could not reach Timestream after version upgrade

## Description
Updates pom file with the new version of ***Flink 1.15.2***, after that change the runtime exception occurs for initialization ***connection: java.lang.NoSuchMethodError: 'org.apache.flink.metrics.MetricGroup org.apache.flink.api.connector.sink.Sink$InitContext.metricGroup()'***

For the fix all pom dependency has been updated with latest aws bom file and Flink dependencies to avoid interface incompatibility
Changes has been added to test cases to support new interface changes

## Related Issue
https://github.com/awslabs/amazon-timestream-tools/issues/122

## Tests performed / created
Run existing unit tests: ***mvn clean test***

### Precondition:
- Build connector: ***mvn clean package***
- Modify pom file for ***"sample-kinesis-to-timestream-app"*** to use the jar of packaged t***imestream-connector***
- Build sample application: ***mvn clean install***
- Run sample application: ***mvn install exec:java -Dexec.mainClass="com.amazonaws.samples.kinesis2timestream.StreamingJob" -Dexec.args="--InputStreamName TimestreamTestStream --Region us-east-1 --TimestreamDbName kdaflink --TimestreamTableName kinesisdata" -Dexec.classpathScope=test***

#### Before changes: 
The connector throw runtime exceptions: ***org.apache.flink.metrics.MetricGroup org.apache.flink.api.connector.sink.Sink$InitContext.metricGroup()***
Connector has not been established

#### After changes:
In the logs no error message and connector initialized, DB and required table created

## Additional Reviewers
@alexeytemnikov

### CODE REVIEW

**Comment by RoyZhang2022** [comment]
Please update PR title to [AT-1456] .....

[AT-1456]: https://bitquill.atlassian.net/browse/AT-1456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java [hunk 1]
```diff
@@ -60,6 +60,11 @@ public int getNumberOfParallelSubtasks() {
         return 0;
     }
 
+    @Override
+    public int getAttemptNumber() {
```
Comment (RoyZhang2022) [inline integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java:64]: Why is this change needed?

**Comment by RoyZhang2022** [comment]
Please update the document to update the command to the below one

```mvn install exec:java -Dexec.mainClass="com.amazonaws.samples.kinesis2timestream.StreamingJob" -Dexec.args="--InputStreamName TimestreamTestStream --Region us-east-1 --TimestreamDbName kdaflink --TimestreamTableName kinesisdata" -Dexec.classpathScope=test```

File: integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java [hunk 2]
```diff
@@ -60,6 +60,11 @@ public int getNumberOfParallelSubtasks() {
...
+        return 1;
```
Comment (alexey-temnikov) [inline integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java]: Why return is constant and always 1?

**Comment by alexey-temnikov** [comment]
Please use following PR template
## PR Template

### Summary


### Description


### Related Issue


### Tests performed / created


### Additional Reviewers
@alexeytemnikov

**Review by alexey-temnikov — DISMISSED** [review]
Dima, please address the comments above.

File: integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java (hunk 1 above)
Comment (dkovalenkoI) [reply integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java:64]: Because we update the version of flink connector code to 1.7 and in that version org.apache.flink.api.connector.sink2.InitContext have new methods that should be implemented getAttemptNumber.

File: integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java (hunk 2 above)
Comment (dkovalenkoI) [reply integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java]: I checked previous version (Before update the flink to 1.7 the default value was 1 for attempts) and in new version we don't have default value.(Need to override interface).

**Comment by RoyZhang2022** [comment]
Please add bracket to `AT-1456` of the title. Please add spaces to the sections of PR description to make it reading friendly.

**Comment by RoyZhang2022** [comment]
Two comments. For the summay I think no need to add aws issue link as it will add our internal info to the aws issue.
The other is for the "related issue", just need to put our Jira link there, no need to link to aws issue.

**Review by RoyZhang2022 — APPROVED** [review]

File: integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java (hunk 2 above)
Comment (alexey-temnikov) [reply integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java]: Let's add a code comment explaining why 1 is a default value.

Comment (dkovalenkoI) [reply integrations/flink_connector/flink-connector-timestream/src/test/java/com/amazonaws/samples/connectors/timestream/SinkInitContext.java]: Comment has been added

**Review by alexey-temnikov — APPROVED** [review]
